### PR TITLE
Fix samples of ExactAssetImage

### LINKED ISSUE #79720: Typo in the ExactAssetImage documentation
There are two occurrences of a typo in the `ExactAssetImage` documentation, where, instead, `AssetImage` was used.

These occurrences are (as of 824b1ad3f862d4c9661f7d1e601d8df8915400a6) in lines 1010 and 1020.

### PR DESCRIPTION
**Pull request by mateusfccp**
Fixed two samples of `ExactAssetImage` where, instead of `ExactAssetImage`, they used `AssetImage`.

Fixes #79720.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by mateusfccp** [comment]
cc @Hixie

File: packages/flutter/lib/src/painting/image_provider.dart [hunk 1]
```diff
@@ -1007,7 +1007,7 @@ class MemoryImage extends ImageProvider<MemoryImage> {
 /// Then, to fetch the image and associate it with scale `1.5`, use:
 ///
 /// ```dart
-/// AssetImage('icons/heart.png', scale: 1.5)
+/// ExactAssetImage('icons/heart.png', scale: 1.5)
```
Comment (goderbauer) [inline packages/flutter/lib/src/painting/image_provider.dart:1010]: This should probably be wrapped in a `{@tool snippet}` directive so the analyzer would have caught that AssetImage is wrong here. Can you try that?

Comment (mateusfccp) [reply packages/flutter/lib/src/painting/image_provider.dart:1010]: Of course!

File: packages/flutter/lib/src/painting/image_provider.dart [hunk 2]
```diff
@@ -1006,19 +1006,19 @@ class MemoryImage extends ImageProvider<MemoryImage> {
 ///
 /// Then, to fetch the image and associate it with scale `1.5`, use:
 ///
-/// ```dart
-/// AssetImage('icons/heart.png', scale: 1.5)
-/// ```
+/// {@tool snippet}
+/// ExactAssetImage('icons/heart.png', scale: 1.5)
```
Comment (goderbauer) [inline packages/flutter/lib/src/painting/image_provider.dart:1010]: you still have to wrap things in ```dart

(please look at other examples of `{@tool snippet}` in our code base)

Comment (mateusfccp) [reply packages/flutter/lib/src/painting/image_provider.dart:1011]: Sorry, I did not pay attention.

**Comment by goderbauer** [comment]
Looks like the analyzer found an issue, can you please fix that: https://github.com/flutter/flutter/pull/79721/checks?check_run_id=2351278038

To make the other tests pass, can you also rebase this to the latest master? Thanks.

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**